### PR TITLE
Pia 3947 fix the feedback sending 

### PR DESCRIPTION
**Pull request by vizaknaidavid**

### CODE REVIEW

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/GiniConfiguration.swift [hunk 1]
```diff
@@ -520,13 +520,11 @@ import GiniBankAPILibrary
      ///   - iban: iban description
      ///   - bic: bic description
      ///   - amountToPay: amountToPay description
-    public func cleanup(paymentRecipient: String, paymentReference: String, paymentPurpose: String, iban: String, bic: String, amountToPay: ExtractionAmount) {
+    public func cleanup(paymentRecipient: String, paymentReference: String, paymentPurpose: String, iban: String, bic: String,  amountToPay: ExtractionAmount) {
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/GiniConfiguration.swift:523]:
```suggestion
    public func cleanup(paymentRecipient: String, paymentReference: String, paymentPurpose: String, iban: String, bic: String, amountToPay: ExtractionAmount) {
```

File: CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/DecimalExtensionTest.swift [hunk 2]
```diff
@@ -0,0 +1,28 @@
+//
+//  File.swift
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/DecimalExtensionTest.swift:2]:
```suggestion
//  DecimalExtensionTests.swift
```

File: CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/DecimalExtensionTest.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+//  Created by David Vizaknai on 04.04.2023.
+//
+
+import XCTest
+
+class DecimalExtensionTests: XCTestCase {
+    func testStringValueFromTwoDecimals() {
+        let decimalValue: Decimal = 8.28
+        let decimalString = decimalValue.stringValue(withDecimalPoint: 2)
+        XCTAssertEqual(decimalString, "8.28", "Decimal string value is not equal to expected string value.")
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/DecimalExtensionTest.swift:14]:
```suggestion
        XCTAssertEqual(decimalString, "8.28", "Decimal string value is equal to expected string value.")
```

File: CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/DecimalExtensionTest.swift [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    func testStringValueFromWholeNumber() {
+        let decimalValue: Decimal = 5
+        let decimalString = decimalValue.stringValue(withDecimalPoint: 2)
+        XCTAssertEqual(decimalString, "5.00", "Decimal string value is not equal to expected string value.")
+    }
+
+    func testStringValueFromOneDecimal() {
+        let decimalValue: Decimal = 210.3
+        let decimalString = decimalValue.stringValue(withDecimalPoint: 2)
+        XCTAssertEqual(decimalString, "210.30", "Decimal string value is not equal to expected string value.")
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/DecimalExtensionTest.swift:26]:
```suggestion
        XCTAssertEqual(decimalString, "210.30", "Decimal string value is equal to expected string value.")
```

File: CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/DecimalExtensionTest.swift [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    func testStringValueFromTwoDecimals() {
+        let decimalValue: Decimal = 8.28
+        let decimalString = decimalValue.stringValue(withDecimalPoint: 2)
+        XCTAssertEqual(decimalString, "8.28", "Decimal string value is not equal to expected string value.")
+    }
+
+    func testStringValueFromWholeNumber() {
+        let decimalValue: Decimal = 5
+        let decimalString = decimalValue.stringValue(withDecimalPoint: 2)
+        XCTAssertEqual(decimalString, "5.00", "Decimal string value is not equal to expected string value.")
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/DecimalExtensionTest.swift:20]:
```suggestion
        XCTAssertEqual(decimalString, "5.00", "Decimal string value is equal to expected string value.")
```

File: CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/DecimalExtensionTest.swift [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+class DecimalExtensionTests: XCTestCase {
+    func testStringValueFromTwoDecimals() {
+        let decimalValue: Decimal = 8.28
+        let decimalString = decimalValue.stringValue(withDecimalPoint: 2)
+        XCTAssertEqual(decimalString, "8.28", "Decimal string value is not equal to expected string value.")
+    }
+
+    func testStringValueFromWholeNumber() {
+        let decimalValue: Decimal = 5
+        let decimalString = decimalValue.stringValue(withDecimalPoint: 2)
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/DecimalExtensionTest.swift:19]: [Suggestion]@vizaknaidavid Could you please add more test cases here for 
1000.09 ?

File: CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/DecimalExtensionTest.swift [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        XCTAssertEqual(decimalString, "5.00", "Decimal string value is not equal to expected string value.")
+    }
+
+    func testStringValueFromOneDecimal() {
+        let decimalValue: Decimal = 210.3
+        let decimalString = decimalValue.stringValue(withDecimalPoint: 2)
+        XCTAssertEqual(decimalString, "210.30", "Decimal string value is not equal to expected string value.")
+    }
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/DecimalExtensionTest.swift:27]:
```suggestion
    }
        func testStringValueFromTwoDecimals2() {
        let decimalValue: Decimal = 200.37
        let decimalString = decimalValue.stringValue(withDecimalPoint: 2)
        XCTAssertEqual(decimalString, "200.37", "Decimal string value is equal to expected string value.")
    }
    
    func testStringValueFromTwoDecimals3() {
        let decimalValue: Decimal = 999.35
        let decimalString = decimalValue.stringValue(withDecimalPoint: 2)
        XCTAssertEqual(decimalString, "999.35", "Decimal string value is equal to expected string value.")
    }
    
    func testStringValueFromTwoDecimals4() {
        let decimalValue: Decimal = 0.09
        let decimalString = decimalValue.stringValue(withDecimalPoint: 2)
        XCTAssertEqual(decimalString, "0.09", "Decimal string value is equal to expected string value.")
    }
```

**Review by zladzeyka — CHANGES_REQUESTED** [review]
Great progress. "paymentReferenceExtraction" a very good catch 👍 
I've added a few small suggestions.

File: CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/DecimalExtensionTest.swift (hunk 4 above)
Comment (vizaknaidavid) [reply CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/DecimalExtensionTest.swift:26]: The string represents the description of the failure according to the Apple documentation. So it should be correct as it is.

**Pull request merged**